public/index.php

```diff
@@ -1,4 +1,6 @@
 <?php
+/* Run install.php, if present*/
+if (file_exists('install.php')) {header("Location: /install.php"); die();}
```

Comment:
First, the user needs to move install.php in place, if he wants to use this install script.

Second, you can put some nice "Do you really want it? dialog" in front of it. I do not care.

It is only executed if install.php is present. If its not, nothing is executed. We can have a long debate, if this is a security issue. Actually, it works without this line in index.php. The user has to use /install.php as the url.

The problem is that you cannot rely on the installation of composer at least not on Ubuntu. So if you want this to be more reliable then you need to install composer automatically during installation.

Actually you did get the point. Installation is about reliability and not about beauty. It does not look nice. It does not need to follow coding guide lines. It needs to work for every single user. Nothing is more frustrating as having trouble with the installation. Since I had this trouble, I decided to do something about it.

So, as usual I do not care if you like my PR or not and I do not care about acceptance. You can take it or leave it. I need it to do my installation since the existing stuff is buggy.

At least you should copy the idea to have a PHP only install.

You have basically two options to do it. You can do either code them in composer.json and use the scripts section. But you should split the tasks in subtasks there. Your install-mechanic can then use composer run-script to execute them. The other option is to code them as php-scripts and run these.

Actually my problem was, that the shell tasks where no executed correctly, but I figured this out when my script was almost complete.

By the way, you should not rely on any shell commands or shell extension for the installation, if you want life easier for Windows users. At least I tried to do that.